common/src/main/java/com/wynntils/models/raid/RaidModel.java

```diff
@@ -94,10 +137,23 @@ public void onTitle(TitleSetTextEvent event) {
         }
     }
 
-    // One challenge in Nexus of Light does not display the scoreboard upon challenge completion
-    // so we have to check for the chat message
     @SubscribeEvent
     public void onChatMessage(ChatMessageReceivedEvent event) {
+        StyledText styledText = event.getOriginalStyledText();
+
+        Matcher rewardPullMatcher = styledText.getMatcher(REWARD_PULLS_PATTERN);
+        if (rewardPullMatcher.find()) {
+            expectedNumRewardPulls = Integer.parseInt(rewardPullMatcher.group(1));
+            return;
+        }
+        Matcher aspectPullMatcher = styledText.getMatcher(ASPECT_PULLS_PATTERN);
+        if (aspectPullMatcher.find()) {
+            expectedNumAspectPulls = Integer.parseInt(aspectPullMatcher.group(1));
+            return;
+        }
+
+        // One challenge in Nexus of Light does not display the scoreboard upon challenge completion so we have to check
+        // for the chat message
```

Comment:
This comment doesn't make sense here, move it down to where the actual completed challenge check is